lib/gpu/command_buffer.cc

```diff
@@ -35,15 +37,18 @@ bool CommandBuffer::Submit() {
   for (auto& encodable : encodables_) {
     encodable->EncodeCommands();
   }
-  return command_buffer_->SubmitCommands();
+
+  return context_->GetCommandQueue()->Submit({command_buffer_}).ok();
```

Comment:
So I'll just need to do this in a raster task to keep it safe, right?

Reply:
this has the exact same threading guarantees as CommandBuffer->Submit. Its the AiksContext batching that is not necessarily thread safe.